api/graphql/reservations/reservation_serializers.py

```diff
@@ -1167,3 +1167,137 @@ class Meta:
+                ValidationErrorCodes.RESERVATION_MODIFICATION_NOT_ALLOWED,
+            )
+        begin = data["begin"]
+        end = data["end"]
+        self.check_begin(begin, end)
+
+        for reservation_unit in self.instance.reservation_unit.all():
+            self.check_cancellation_rules(reservation_unit)
+            self.check_reservation_time(reservation_unit, begin, end)
+            self.check_reservation_overlap(reservation_unit, begin, end)
+            self.check_reservation_duration(reservation_unit, begin, end)
+            self.check_buffer_times(reservation_unit, begin, end)
+            self.check_reservation_days_before(begin, reservation_unit)
+
+            scheduler = ReservationUnitReservationScheduler(
+                reservation_unit, opening_hours_end=end.date()
+            )
+            self.check_opening_hours(scheduler, begin, end)
+            self.check_open_application_round(scheduler, begin, end)
+            self.check_reservation_start_time(scheduler, begin)
+
+        self.check_and_handle_pricing(data)
+
+        if self.instance._requires_handling():
+            data["state"] = STATE_CHOICES.REQUIRES_HANDLING
+
+        return data
+
+    def check_begin(self, begin, end):
+        if begin > end:
+            raise ValidationErrorWithCode(
+                "End cannot be before begin",
+                ValidationErrorCodes.RESERVATION_MODIFICATION_NOT_ALLOWED,
+            )
+
+        now = datetime.datetime.now(tz=DEFAULT_TIMEZONE)
+
+        if begin < now:
+            raise ValidationErrorWithCode(
+                "Reservation new begin cannot be in the past",
+                ValidationErrorCodes.RESERVATION_BEGIN_IN_PAST,
+            )
+
+        if self.instance.begin < now:
+            raise ValidationErrorWithCode(
+                "Reservation time cannot be changed when current begin time is in past.",
+                ValidationErrorCodes.RESERVATION_CURRENT_BEGIN_IN_PAST,
+            )
+
+    def check_cancellation_rules(self, reservation_unit):
+        now = datetime.datetime.now(tz=DEFAULT_TIMEZONE)
+
+        cancel_rule = reservation_unit.cancellation_rule
+        if not cancel_rule:
+            raise ValidationErrorWithCode(
+                "Reservation time cannot be changed thus no cancellation rule.",
+                ValidationErrorCodes.CANCELLATION_NOT_ALLOWED,
+            )
+        must_be_cancelled_before = (
+            self.instance.begin - cancel_rule.can_be_cancelled_time_before
+        )
+        if must_be_cancelled_before < now:
+            raise ValidationErrorWithCode(
+                "Reservation time cannot be changed because the cancellation period has expired.",
+                ValidationErrorCodes.CANCELLATION_NOT_ALLOWED,
+            )
+        if cancel_rule.needs_handling:
+            raise ValidationErrorWithCode(
+                "Reservation time change needs manual handling.",
```

Comment:
I'd change the message to be more explaining. Something like:
```suggestion
                "Reservation time cannot be changed because cancel rule needs manual handling.",
```